python/main_loop.py

```diff
@@ -148,13 +148,16 @@ def createNewLocalPath(sailbot, maxAllowableRuntimeSeconds, desiredHeadingPublis
                 sailbot.newGlobalPathReceived = False
 
             # If globalWaypoint reached, increment the index
-            if isGlobalWaypointReached or goalInvalid:
+            if isGlobalWaypointReached:
                 if isGlobalWaypointReached:
                     rospy.loginfo("Global waypoint reached")
-                if goalInvalid:
-                    rospy.logwarn("Goal state invalid, skipping to the next global waypoint")
                 sailbot.globalPathIndex += 1
 
+            # If goal invalid, use a new global waypoint
```

Comment:
Great comment, could add at the end "move global waypoint toward/away from the sailbot until it is valid" to be extra clear